src/System.ServiceModel.UnixDomainSocket/src/System/ServiceModel/UnixDomainSocketSecurity.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Diagnostics.Contracts;
+using System.ServiceModel.Channels;
+
+namespace System.ServiceModel
+{
+    public sealed class UnixDomainSocketSecurity
+    {
+        internal const SecurityMode DefaultMode = SecurityMode.Transport;
+
+        private SecurityMode _mode;
```

Comment:
Need to create a UDS specific enum for this as we aren't supporting Message or TransportWithMessageCredentials. See [NetNamedPipeSecurityMode Enum docs.](https://learn.microsoft.com/en-us/dotnet/api/system.servicemodel.netnamedpipesecuritymode?view=dotnet-plat-ext-7.0)

Reply:
This also goes with [my comment here](https://github.com/dotnet/wcf/pull/5172#issuecomment-1749514065) where we need TransportCredentialOnly for IdentityOnly